### PR TITLE
Makefile: Update Cockpit lib to 756b59469637dd2dee00d086b883cffd16dc5efc

### PR DESCRIPTION
**Pull request by martinpitt**
Update to the same PF5 versions as Cockpit. Apply the "-v5" prefixing and import adjustments.

Make use of the new pywrap feature from our test.

See starter-kit commit 2215aa3bf801224.

---

Copy of PR #1299, as that has a broken GitHub API.

### CODE REVIEW

**Comment by martinpitt** [comment]
Bah, needs node_modules refresh.

File: src/ImageRunModal.jsx [hunk 1]
```diff
@@ -652,7 +652,7 @@ export class ImageRunModal extends React.Component {
         // Add the search component
         const footer = (
             <ToggleGroup className='image-search-footer' aria-label={_("Search by registry")}>
-                <ToggleGroupItem text={_("All")} key='all' isSelected={this.state.searchByRegistry == 'all'} onChange={(_, ev) => {
+                <ToggleGroupItem text={_("All")} key='all' isSelected={this.state.searchByRegistry == 'all'} onChange={(ev, _) => {
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:655]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 2]
```diff
@@ -735,7 +735,7 @@ export class ImageRunModal extends React.Component {
                                                   </TextContent>
                                               </>
                                           }>
-                                           <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                           <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:738]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 3]
```diff
@@ -765,7 +765,7 @@ export class ImageRunModal extends React.Component {
                                         <FlexItem>{cockpit.format(_("Searching: $0"), "quay.io/busybox")}</FlexItem>
                                     </Flex>
                                 }>
-                                  <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                  <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:768]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 4]
```diff
@@ -861,7 +861,7 @@ export class ImageRunModal extends React.Component {
                                   <Popover aria-label={_("CPU Shares help")}
                                       enableFlip
                                       bodyContent={_("CPU shares determine the priority of running containers. Default priority is 1024. A higher number prioritizes this container. A lower number decreases priority.")}>
-                                      <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:864]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 5]
```diff
@@ -893,7 +893,7 @@ export class ImageRunModal extends React.Component {
                               <Popover aria-label={_("Restart policy help")}
                                 enableFlip
                                 bodyContent={this.props.userLingeringEnabled ? _("Restart policy to follow when containers exit. Using linger for auto-starting containers may not work in some circumstances, such as when ecryptfs, systemd-homed, NFS, or 2FA are used on a user account.") : _("Restart policy to follow when containers exit.")}>
-                                  <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                  <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:896]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 6]
```diff
@@ -960,19 +960,19 @@ export class ImageRunModal extends React.Component {
...
+                           onChange={(_, value) => this.onValueChanged('healthcheck_command', value)} />
                         </FormGroup>
 
                         <FormGroup fieldId='run-image-healthcheck-interval' label={_("Interval")}
                               labelIcon={
                                   <Popover aria-label={_("Health check interval help")}
                                       enableFlip
                                       bodyContent={_("Interval how often health check is run.")}>
-                                      <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:975]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 7]
```diff
@@ -997,7 +997,7 @@ export class ImageRunModal extends React.Component {
                                   <Popover aria-label={_("Health check timeout help")}
                                       enableFlip
                                       bodyContent={_("The maximum time allowed to complete the health check before an interval is considered failed.")}>
-                                      <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:1000]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 8]
```diff
@@ -1022,7 +1022,7 @@ export class ImageRunModal extends React.Component {
                                   <Popover aria-label={_("Health check start period help")}
                                       enableFlip
                                       bodyContent={_("The initialization time needed for a container to bootstrap.")}>
-                                      <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:1025]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 9]
```diff
@@ -1047,7 +1047,7 @@ export class ImageRunModal extends React.Component {
                                   <Popover aria-label={_("Health check retries help")}
                                       enableFlip
                                       bodyContent={_("The number of retries allowed before a healthcheck is considered to be unhealthy.")}>
-                                      <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:1050]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx [hunk 10]
```diff
@@ -1070,7 +1070,7 @@ export class ImageRunModal extends React.Component {
                                   <Popover aria-label={_("Health failure check action help")}
                                       enableFlip
                                       bodyContent={_("Action to take once the container transitions to an unhealthy state.")}>
-                                      <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/ImageRunModal.jsx:1073]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/PublishPort.jsx [hunk 11]
```diff
@@ -20,14 +20,14 @@ export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount })
                     <Popover aria-label={_("IP address help")}
                         enableFlip
                         bodyContent={_("If host IP is set to 0.0.0.0 or not set at all, the port will be bound on all IPs on the host.")}>
-                        <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                        <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/PublishPort.jsx:23]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/PublishPort.jsx [hunk 12]
```diff
@@ -36,7 +36,7 @@ export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount })
                         <Popover aria-label={_("Host port help")}
                             enableFlip
                             bodyContent={_("If the host port is not set the container port will be randomly assigned a port on the host.")}>
-                            <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                            <button onClick={e => e.preventDefault()} className="pf-v5-c-form__group-label-help">
```
Comment (cockpituous) [inline src/PublishPort.jsx:39]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
The flakes here are horrible. I retry a few times. @jelly , I assume in #1299 you also retried several times? I'll have a closer look today, this is unbearable.

But wrt. this PF5 update, [this screenshot](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel/TestApplication-testCreatePodUser-fedora-38-127.0.0.2-2501-FAIL.png) looks broken. The CPU and memory icons are surely not supposed to be that big? Note that PF5 alpha increased all the sizes by one by default.

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:655]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 2 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:738]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 3 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:768]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 4 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:864]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 5 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:896]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 6 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:975]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 7 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1000]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 8 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1025]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 9 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1050]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 10 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1073]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/PublishPort.jsx (hunk 11 above)
Comment (cockpituous) [inline src/PublishPort.jsx:23]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/PublishPort.jsx (hunk 12 above)
Comment (cockpituous) [inline src/PublishPort.jsx:39]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-044046-64a79ae1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
A common failure is `TestApplication.testNotRunning`, which fails on all OSes with the Python bridge, like [arch](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-arch/log.html#21-1) or [debian-testing](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-debian-testing/log.html#21). I sent a proper fix for this to https://github.com/cockpit-project/cockpit/pull/19052

I added another commit here to work around the messages. I have some difficulty reproducing this locally, so validating with the bots (I might have forgotten a particular log line)

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:655]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 2 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:738]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 3 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:768]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 4 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:864]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 5 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:896]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 6 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:975]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 7 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1000]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 8 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1025]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 9 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1050]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/ImageRunModal.jsx (hunk 10 above)
Comment (cockpituous) [inline src/ImageRunModal.jsx:1073]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/PublishPort.jsx (hunk 11 above)
Comment (cockpituous) [inline src/PublishPort.jsx:23]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/PublishPort.jsx (hunk 12 above)
Comment (cockpituous) [inline src/PublishPort.jsx:39]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-053724-7cc9e4bf-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
I got [this testCreatePodUser error](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-042644-64a79ae1-fedora-38-devel/log.html#12-1) locally once when I was investigating something else. But of course since then I could never reproduce it again. The pod container is in state "Created", while it's supposed to be "Running". That isn't just a timeout question, it stays in "Created" forever. This is very likely related to

> warning: Failed to do Update Pod: {"problem":"disconnected","message":"Server has closed the connection."}

The journal says

> Jul 03 04:31:04 fedora-38-127-0-0-2-2201 podman[7848]: 2023-07-03 04:31:04.901937656 +0000 UTC m=+0.349787907 container start f8f32ccac07676234bf0e1bd85e045ee95d0bd9e615b1be9206758091e5f4b8d (image=localhost/test-alpine:latest, name=test-pod-1....]
> Jul 03 04:31:08 fedora-38-127-0-0-2-2201 podman[7981]: 2023-07-03 04:31:08.009154762 +0000 UTC m=+0.053553284 container died f8f32ccac07676234bf0e1bd85e045ee95d0bd9e615b1be9206758091e5f4b8d (image=localhost/test-alpine:latest, name=test-pod-1, 

so for some reason the container itself has given up? But it's a bit hard to imagine how a `sleep 500` could die in 4 seconds.. There's nothing more helpful in the journal in between these.

I tried to amplify this with the patch below, but it seems stable for me now.  That condition is triggered by something else..

```diff
diff --git test/check-application test/check-application    
index d0b65be4..c10c1c4a 100755    
--- test/check-application    
+++ test/check-application    
@@ -2610,6 +2610,15 @@ class TestApplication(testlib.MachineCase):     
·    
         container_name = 'test-pod-1-system' if auth else 'test-pod-1'    
         cmd = f"podman run -d --pod {pod_name} --name {container_name} --stop-timeout 0 {IMG_ALPINE} sleep 500"    
+    
+        for _ in range(50):    
+            containerId = self.execute(auth, cmd).strip()    
+            self.waitPodContainer(pod_name,    
+                                [{"name": container_name, "image": IMG_ALPINE,    
+                                    "command": "sleep 500", "state": "Running", "id": containerId}], auth)    
+            self.execute(auth, f"podman rm -f {container_name}")    
+            self.waitPodContainer(pod_name, [])    
+    
         containerId = self.execute(auth, cmd).strip()    
         self.waitPodContainer(pod_name,    
                               [{"name": container_name, "image": IMG_ALPINE,    
```

**Comment by martinpitt** [comment]
[This testPruneUnusedContainersUser failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1322-20230703-061631-7cc9e4bf-rhel4edge/log.html#28-1) is a variation of https://github.com/cockpit-project/bots/issues/4829 . I sent https://github.com/cockpit-project/bots/pull/4957 to fix this.

**Review by jelly — APPROVED** [review]

**Pull request merged**